### PR TITLE
I created a program that creates an array of half random shapes with …

### PR DESCRIPTION
**Pull request by IDMFokin**
…a random color, and half by default with a white krg with a radio frame of 10 units.

### CODE REVIEW

File: src/main/java/core/basesyntax/Main.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package core.basesyntax;
+
+import core.basesyntax.figure.Figure;
+import core.basesyntax.supplier.FigureSupplier;
+
+public class Main {
+    private static final int VALUE = 2;
```
Comment (Nick97-git) [inline src/main/java/core/basesyntax/Main.java:7]: it's a redundant constant, let's remove it

File: src/main/java/core/basesyntax/Main.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+    public static void main(String[] args) {
+        Figure [] figures = new Figure[6];
```
Comment (Nick97-git) [inline src/main/java/core/basesyntax/Main.java:10]: create constant from 6

File: src/main/java/core/basesyntax/Main.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        FigureSupplier figureSupplier = new FigureSupplier();
+
```
Comment (Nick97-git) [inline src/main/java/core/basesyntax/Main.java:12]: remove empty line

File: src/main/java/core/basesyntax/figure/Circle.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
+package core.basesyntax.figure;
+
+import core.basesyntax.interfaces.AreaCalculator;
+
+public class Circle extends Figure implements AreaCalculator {
```
Comment (Nick97-git) [inline src/main/java/core/basesyntax/figure/Circle.java:5]:
```suggestion
public class Circle extends Figure {
```
the same for other figures

File: src/main/java/core/basesyntax/figure/Circle.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    private static final String TYPE_FIGURE = "CIRCLE";
```
Comment (Nick97-git) [inline src/main/java/core/basesyntax/figure/Circle.java:6]: you don't need such constant, remove it, the same for other figures, you can use directly in method enum value instead

File: src/main/java/core/basesyntax/figure/Figure.java [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+
+    protected Figure(String color) {
+        this.color = color;
+    }
+
+    protected String getColor() {
+        return color;
+    }
+
+    public abstract void print();
```
Comment (Nick97-git) [inline src/main/java/core/basesyntax/figure/Figure.java:14]: let's move this method to the separate interface Drawable and rename print() to draw(), Figure should implement this interface

File: src/main/java/core/basesyntax/figure/IsoscelesTrapezoid.java [hunk 7]
```diff
@@ -0,0 +1,55 @@
+package core.basesyntax.figure;
+
+import core.basesyntax.interfaces.AreaCalculator;
+
+public class IsoscelesTrapezoid extends Figure implements AreaCalculator {
+    private static final String TYPE_FIGURE = "ISOSCELES_TRAPEZOID";
+    private static final double NUMBER = 2.0;
```
Comment (Nick97-git) [inline src/main/java/core/basesyntax/figure/IsoscelesTrapezoid.java:7]: it's a redundant constant, remove it

File: src/main/java/core/basesyntax/figure/RightTriangle.java [hunk 8]
```diff
@@ -0,0 +1,45 @@
+package core.basesyntax.figure;
+
+import core.basesyntax.interfaces.AreaCalculator;
+
+public class RightTriangle extends Figure implements AreaCalculator {
+    private static final String TYPE_FIGURE = "RIGHT_TRIANGLE";
+    private static final double NUMBER = 2.0;
```
Comment (Nick97-git) [inline src/main/java/core/basesyntax/figure/RightTriangle.java:7]: it's a redundant constant, remove it

File: src/main/java/core/basesyntax/interfaces/AreaCalculator.java [hunk 9]
```diff
@@ -0,0 +1,6 @@
+package core.basesyntax.interfaces;
+
+public interface AreaCalculator {
+    double area();
+
```
Comment (Nick97-git) [inline src/main/java/core/basesyntax/interfaces/AreaCalculator.java:5]: remove empty line

File: src/main/java/core/basesyntax/interfaces/AreaCalculator.java [hunk 10]
```diff
@@ -0,0 +1,6 @@
+package core.basesyntax.interfaces;
+
+public interface AreaCalculator {
+    double area();
```
Comment (Nick97-git) [inline src/main/java/core/basesyntax/interfaces/AreaCalculator.java:4]:
```suggestion
    double getArea();
```

File: src/main/java/core/basesyntax/supplier/FigureSupplier.java [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+import core.basesyntax.figure.Rectangle;
+import core.basesyntax.figure.RightTriangle;
+import core.basesyntax.figure.Square;
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int MAX_VALUE = 30;
+    private static final int DEFAULT_RADIUS = 10;
+    private final Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
```
Comment (Nick97-git) [inline src/main/java/core/basesyntax/supplier/FigureSupplier.java:15]:
```suggestion
    private static final int MAX_RANDOM_VALUE = 30;
    private static final int DEFAULT_CIRCLE_RADIUS = 10;
    private final Random random = new Random();
    private final ColorSupplier colorSupplier = new ColorSupplier();
```

File: src/main/java/core/basesyntax/supplier/FigureSupplier.java [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+
+    public Figure getRandomFigure() {
+        int randomIndex = random.nextInt(FigureType.values().length);
+        String color;
```
Comment (Nick97-git) [inline src/main/java/core/basesyntax/supplier/FigureSupplier.java:19]:
```suggestion
        String color = colorSupplier.getRandomColor();
```

File: src/main/java/core/basesyntax/supplier/FigureSupplier.java [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+        double sideA;
+        double sideB;
+
+        switch (randomIndex) {
+            case 0:
+                color = colorSupplier.getRandomColor();
```
Comment (Nick97-git) [inline src/main/java/core/basesyntax/supplier/FigureSupplier.java:25]: remove such line from all cases

**Review by Nick97-git — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/interfaces/AreaCalculator.java (hunk 10 above)
Comment (IDMFokin) [reply src/main/java/core/basesyntax/interfaces/AreaCalculator.java:4]: Tell me please. You said above. I need to delete interface public class Circle extends Figure implements AreaCalculator.
I need to rename the method and move it to abstract class?

File: src/main/java/core/basesyntax/Main.java [hunk 14]
```diff
@@ -0,0 +1,23 @@
+package core.basesyntax;
+
+import core.basesyntax.figure.Figure;
+import core.basesyntax.supplier.FigureSupplier;
+
+public class Main {
+    private static final int ARRAY_SIZE = 6;
```
Comment (Nick97-git) [inline src/main/java/core/basesyntax/Main.java:7]:
```suggestion
    private static final int FIGURES_SIZE = 6;
```

File: src/main/java/core/basesyntax/figure/Figure.java (hunk 6 above)
Comment (Nick97-git) [reply src/main/java/core/basesyntax/figure/Figure.java:16]: what about this comment?

File: src/main/java/core/basesyntax/figure/IsoscelesTrapezoid.java (hunk 7 above)
Comment (Nick97-git) [reply src/main/java/core/basesyntax/figure/IsoscelesTrapezoid.java:6]: what about this comment?

File: src/main/java/core/basesyntax/figure/RightTriangle.java (hunk 8 above)
Comment (Nick97-git) [reply src/main/java/core/basesyntax/figure/RightTriangle.java:6]: what about this comment?

File: src/main/java/core/basesyntax/supplier/FigureType.java [hunk 15]
```diff
@@ -0,0 +1,9 @@
+package core.basesyntax.supplier;
+
+public enum FigureType {
```
Comment (Nick97-git) [inline src/main/java/core/basesyntax/supplier/FigureType.java:3]: let's remove this enum

File: src/main/java/core/basesyntax/supplier/FigureSupplier.java [hunk 16]
```diff
@@ -0,0 +1,49 @@
...
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int MAX_RANDOM_VALUE = 30;
+    private static final int DEFAULT_CIRCLE_RADIUS = 10;
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int randomIndex = random.nextInt(FigureType.values().length);
```
Comment (Nick97-git) [inline src/main/java/core/basesyntax/supplier/FigureSupplier.java:18]: let's use inside of nextInt constant that will have a number of available figures as its value

File: src/main/java/core/basesyntax/supplier/FigureSupplier.java [hunk 17]
```diff
@@ -0,0 +1,49 @@
...
+        String color = colorSupplier.getRandomColor();;
+        double sideA;
+        double sideB;
+
```
Comment (Nick97-git) [inline src/main/java/core/basesyntax/supplier/FigureSupplier.java:22]: remove empty line

**Review by Nick97-git — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/interfaces/AreaCalculator.java (hunk 10 above)
Comment (Nick97-git) [reply src/main/java/core/basesyntax/interfaces/AreaCalculator.java]: you've done right

**Pull request closed without merging**